spinalcordtoolbox/metrics.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# Functions dealing with metrics quantification across slices and/or vertebral levels and output (csv, etc.)
+
+import numpy as np
+import sct_utils as sct
+from sct_image import Image, set_orientation
+from spinalcordtoolbox.utils import parse_num_list
+from spinalcordtoolbox.template import get_slices_from_vertebral_levels
+
+
+def average_per_slice_or_level(metrics, header='', slices='', perslice=1, vert_levels='', perlevel=0,
+                               fname_vert_levels='', file_out='metrics', overwrite=1):
+
+    # TODO: combine metrics+header into an OrderedDictionary
```

Comment:
the rationale behind OrderedDictionary was to be able to sort keys for easier output as a csv (because we want to follow a specific key order).

Reply:
I'd leave that ordering responsibility to the calling code, to avoid having to know about it in callees. Eg. if we return the aggregated metrics, we'd also have to use an OrderedDict...